dm/pkg/retry/errors.go

```diff
@@ -100,3 +101,24 @@ func IsUnretryableConnectionError(err error) bool {
 	// If the last write isn't idempotent, retry it may cause problems.
 	return errors.Cause(err) == dmysql.ErrInvalidConn
 }
+
+// IsRetryableConnectionErrorOnBegin checks whether an invalid connection error occured on a
+// transaction begin statement, when there are no operations that may need to be rolled back.
+func IsRetryableConnectionErrorOnBegin(err error) bool {
+	if !strings.Contains(strings.ToLower(err.Error()), "execute statement failed: begin") {
```

Comment:
Seems it's generated from here

https://github.com/pingcap/tiflow/blob/e23fa78655fa5122430dc92fed1be6f0a827c725/dm/pkg/conn/baseconn.go#L168

Maybe you can create a special error variable called `ErrDBExecuteFailedBegin` and use `terror.ErrDBExecuteFailedBegin.Equal(err)` to do a more structured check. The current way of check substring in error message will fail after another developer accidentally change the message.

Reply:
Thanks, this is a good idea!